aten/src/ATen/native/layer_norm.cpp

```diff
@@ -186,6 +189,71 @@ Tensor layer_norm(
   return std::get<0>(at::native_layer_norm(input, normalized_shape, weight, bias, eps));
 }
 
+Tensor& layer_norm_out(
+    const Tensor& input,
+    IntArrayRef normalized_shape,
+    const c10::optional<Tensor>& weight_opt /* optional */,
+    const c10::optional<Tensor>& bias_opt /* optional */,
+    double eps,
+    bool /* cudnn_enable, deprecated */,
+    Tensor& output) {
+  c10::MaybeOwned<Tensor> weight_maybe_owned =
+      at::borrow_from_optional_tensor(weight_opt);
+  const Tensor& weight = *weight_maybe_owned;
+  c10::MaybeOwned<Tensor> bias_maybe_owned =
+      at::borrow_from_optional_tensor(bias_opt);
+  const Tensor& bias = *bias_maybe_owned;
+
+  auto M_N = _check_layer_norm_inputs(input, normalized_shape, weight, bias);
+  auto M = M_N.first;
+  auto N = M_N.second;
+  auto X = input.expect_contiguous();
+  auto gamma = weight.expect_contiguous();
+  auto beta = bias.expect_contiguous();
+
+  TORCH_CHECK(canCast(
+      typeMetaToScalarType(input.dtype()),
+      typeMetaToScalarType(output.dtype())));
+
+  if (!output.is_contiguous()) {
+    Tensor Y = at::native::empty_like(
+        *X,
+        c10::nullopt /* dtype */,
+        c10::nullopt /* layout */,
+        c10::nullopt /* device */,
+        c10::nullopt /* pin_memory */,
+        at::MemoryFormat::Contiguous);
+    LayerNormKernel(
+        input.device().type(),
+        *X,
+        *gamma,
+        *beta,
+        M,
+        N,
+        eps,
+        &Y,
+        /*mean=*/nullptr,
+        /*rstd=*/nullptr);
```

Comment:
Even if you don't need to compute these (since out variants don't support autograd), the CUDA kernel still assumes that something is passed for them:
https://github.com/pytorch/pytorch/blob/bbb1a55d9f254ad5f587af290b2c975e6f505af9/aten/src/ATen/native/cuda/layer_norm_kernel.cu#L680-L681

I think this explains the segfaults you're getting in the failing CI checks. Note that the out-of-place CPU kernel does this that you can probably copy:
```cpp
Tensor mean = at::empty({M}, X->options());
Tensor rstd = at::empty({M}, X->options());
```

Reply:
I see thanks :)